manifests/plugin/aggregation.pp

```diff
@@ -16,5 +16,10 @@
     'ensure' => $ensure,
   }
 
-  create_resources(collectd::plugin::aggregation::aggregator, $aggregators, $defaults)
+  $aggregators.each |String $resource, Hash $attributes| {
+    Resource['collectd::plugin::aggregation::aggregator'] {
```

Comment:
Would this form be more readable?
```puppet
collectd::plugin::aggregation::aggregator {$resource:
  * => $defaults + $attributes,
}
```

Reply:
Or do away with $defaults and then just
```puppet
collectd::plugin::aggregation::aggregator {$resource:
  ensure => $ensure,
  *      => $attributes,
}
```

Reply:
I agree with @alexjfisher here. I think his first option is shorter and easier to read.

Reply:
> Or do away with $defaults and then just
> 
> ```puppet
> collectd::plugin::aggregation::aggregator {$resource:
>   ensure => $ensure,
>   *      => $attributes,
> }
> ```

this one won't merge, so we are assuming that attributes doesn't contain ensure